icu4c/source/common/uprops.cpp

```diff
@@ -434,15 +542,29 @@ static int32_t getTrailCombiningClass(const IntProperty &/*prop*/, UChar32 c, UP
 #endif
 
 static int32_t getInPC(const IntProperty &, UChar32 c, UProperty) {
-    return ucptrie_get(&inpc_trie, c);
+    return ulayout_ensureData() && gInpcTrie != nullptr ? ucptrie_get(gInpcTrie, c) : 0;
 }
 
 static int32_t getInSC(const IntProperty &, UChar32 c, UProperty) {
-    return ucptrie_get(&insc_trie, c);
+    return ulayout_ensureData() && gInscTrie != nullptr ? ucptrie_get(gInscTrie, c) : 0;
 }
 
 static int32_t getVo(const IntProperty &, UChar32 c, UProperty) {
-    return ucptrie_get(&vo_trie, c);
+    return ulayout_ensureData() && gVoTrie != nullptr ? ucptrie_get(gVoTrie, c) : 0;
+}
+
+static int32_t layoutGetMaxValue(const IntProperty &/*prop*/, UProperty which) {
```

Comment:
No particular reason. I guess I just modified what was there.
I could combine them but would need a switch (property) to pick the trie, then do the nullptr check and lookup. Fewer functions but same or more code.
Less dispatch is better for value lookup? Getting the max value is not a common operation.
Although, these trie lookups via getter functions are not super-optimized either.

Reply:
Maybe if you add more of these tries in the future, you can combine the functions.  It seems nicer since you have the enum already to switch on the enum for nice parallelism between the get-trie and get-max-value.